app/src/androidTest/java/com/duckduckgo/app/browser/SpecialUrlDetectorImplTest.kt

```diff
@@ -59,6 +87,72 @@ class SpecialUrlDetectorImplTest {
         assertEquals("https://example.com", type.webAddress)
     }
 
+    @Test
+    fun whenAppLinksEnabledAndNoNonBrowserActivitiesFoundThenReturnWebType() {
+        whenever(mockSettingsDataStore.appLinksEnabled).thenReturn(true)
+        whenever(mockPackageManager.queryIntentActivities(any(), anyInt())).thenReturn(listOf(buildBrowserResolveInfo()))
+        val type = testee.determineType("https://example.com")
+        assertTrue(type is Web)
+    }
+
+    @Test
+    fun whenAppLinksDisabledThenReturnWebType() {
+        whenever(mockSettingsDataStore.appLinksEnabled).thenReturn(false)
+        whenever(mockPackageManager.queryIntentActivities(any(), anyInt())).thenReturn(listOf(buildAppResolveInfo()))
+        val type = testee.determineType("https://example.com")
+        verifyZeroInteractions(mockPackageManager)
+        assertTrue(type is Web)
+    }
+
+    @Test
+    fun whenAppLinkThrowsURISyntaxExceptionThenReturnWebType() {
+        whenever(mockSettingsDataStore.appLinksEnabled).thenReturn(true)
+        given(mockPackageManager.queryIntentActivities(any(), anyInt())).willAnswer { throw URISyntaxException("", "") }
+        val type = testee.determineType("https://example.com")
+        assertTrue(type is Web)
+    }
+
+    @Test
+    fun whenAppLinksEnabledAndOneNonBrowserActivityFoundThenReturnAppLinkWithIntent() {
+        whenever(mockSettingsDataStore.appLinksEnabled).thenReturn(true)
+        whenever(mockPackageManager.queryIntentActivities(any(), anyInt())).thenReturn(listOf(buildAppResolveInfo(), buildBrowserResolveInfo(), ResolveInfo()))
+        val type = testee.determineType("https://example.com")
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
```

Comment:
how is this going to be tested? depends on the machine / emulator running the tests?

Reply:
We're only handling app links on API 24+ (On API <24 app links will be opened in the browser). So yes, depending on the API of the emulator running the tests, this test will have different behavior.

Reply:
makes sense, should we have two different tests then that only run in a specific target sdk?

Reply:
I'll do that. Splitting it up will make more sense, thanks for the suggestion 👍